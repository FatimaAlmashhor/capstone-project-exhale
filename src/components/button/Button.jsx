import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

export default function Button({ submit, height, width, color, hover, textColor }) {
  const [style, setStyle] = useState('');
  useEffect(() => {
    setStyle(
      `text-${textColor} bg-${color} hover:bg-${hover} border  hover:border-${color}  font-bold py-2 px-3 w-${width} h-${height} transition duration-500  ease-in-out rounded`
    );
  }, []);
  return (
    <div>
      <button type={submit ? 'submit' : 'button'} className={style}>
        Button
      </button>
    </div>
  );
}
Button.propTypes = {
  submit: PropTypes.bool.isRequired,
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  hover: PropTypes.string.isRequired,
};
