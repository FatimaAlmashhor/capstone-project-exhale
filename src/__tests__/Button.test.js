import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/button/Button';

it('renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
