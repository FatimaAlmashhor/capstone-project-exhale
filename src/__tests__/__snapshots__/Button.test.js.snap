// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="  px-8 py-1 sm:px-8 sm:py-1 md:px-10 md:py-2 lg:px-16 lg:py-3 bg-main-pink hover:bg-transparent  border-2  hover:text-main-pink border-main-pink transition duration-500  ease-in-out rounded"
  disabled={false}
  type="button"
/>
`;
